### PR TITLE
Census tools fix

### PR DESCRIPTION
**Pull request by Abilityguy**
This PR makes changes to the US census subject table scripts. The changes can be summarized as

- Minor documentation fixes to `scripts/us_census/acs5yr/subject_tables/common/README.md`
- Removing unused modules and minor lint fixes to `generate_col_map.py`, `data_loader.py`, `common_util_test.py`, `column_map_validator.py`, `column_map_validator_test.py`, `acs_spec_validator.py`, `acs_spec_validator_test.py`
- Renaming S1603 and S2303 specs to `S1603_spec.json` and `S2303_spec.json` as per convention.
- Code and lint fixes to `acs_spec_generator.py`. The script now creates the combined spec using the spec files in each subject table folder. 
- Tests for the spec generator `acs_spec_generator_test.py`. Test artifacts include `common/testdata/expected_generated_spec.json`, `common/testdata/S2303_alabama_2019.zip`, and all files in `common/prefetched_outputs/`

### CODE REVIEW

File: scripts/us_census/acs5yr/subject_tables/common/dc_utils.py [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+_FLAGS = flags.FLAGS
+
+flags.DEFINE_string('dcid', None, 'dcid of the node to query')
+flags.DEFINE_string('dc_output_path', './prefetched_outputs/',
+                    'Path to store the output')
+flags.DEFINE_boolean('force_fetch', False,
+                     'forces api query and not return cached result')
+
+
+def request_url_json(url: str) -> dict:
```
Comment (ajaits) [inline scripts/us_census/acs5yr/subject_tables/common/dc_utils.py:35]: Can this be used instead?
https://github.com/datacommonsorg/data/blob/d4d2f8600ce947c436e8ad630e07d769f2a0cf87/tools/download_utils/requests_wrappers.py#L24

File: scripts/us_census/acs5yr/subject_tables/common/dc_utils.py [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+
+    if req.status_code == requests.codes.ok:
+        response_data = req.json()
+    else:
+        response_data = {'http_err_code': req.status_code}
+        print('HTTP status code: ' + str(req.status_code))
+    return response_data
+
+
+def request_post_json(url: str, data_: dict) -> dict:
```
Comment (ajaits) [inline scripts/us_census/acs5yr/subject_tables/common/dc_utils.py:64]: Can this be used instead?
https://github.com/datacommonsorg/data/blob/d4d2f8600ce947c436e8ad630e07d769f2a0cf87/tools/download_utils/requests_wrappers.py#L54

File: scripts/us_census/acs5yr/subject_tables/common/dc_utils.py [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+    if req.status_code == requests.codes.ok:
+        response_data = req.json()
+    else:
+        response_data = {'http_err_code': req.status_code}
+        print('HTTP status code: ' + str(req.status_code))
+    return response_data
+
+
+# fetch pvs from dc, enums from dc
+def fetch_dcid_properties_enums(class_dcid,
```
Comment (ajaits) [inline scripts/us_census/acs5yr/subject_tables/common/dc_utils.py:94]: I think this was moved here:
https://github.com/datacommonsorg/data/blob/0f069ed0327f95e3f821731aae0d6a00a2565f78/scripts/us_census/acs5yr/subject_tables/common/datacommons_api_wrappers/datacommons_wrappers.py#L99

File: scripts/us_census/acs5yr/subject_tables/common/prefetched_outputs/Person_dc_props.json [hunk 4]
```diff
@@ -0,0 +1,128 @@
+{
```
Comment (ajaits) [inline scripts/us_census/acs5yr/subject_tables/common/prefetched_outputs/Person_dc_props.json:1]: Can prefetched_outputs be generated by the script? If so, perhaps skip these files or add instruction on how to generate them.

**Review by ajaits — COMMENTED** [review]
Thank you Anush for cleaning this up and adding tests :)

I think dc_utils.py was moved to [data](https://github.com/datacommonsorg/data/tree/0f069ed0327f95e3f821731aae0d6a00a2565f78)/[scripts](https://github.com/datacommonsorg/data/tree/0f069ed0327f95e3f821731aae0d6a00a2565f78/scripts)/[us_census](https://github.com/datacommonsorg/data/tree/0f069ed0327f95e3f821731aae0d6a00a2565f78/scripts/us_census)/[acs5yr](https://github.com/datacommonsorg/data/tree/0f069ed0327f95e3f821731aae0d6a00a2565f78/scripts/us_census/acs5yr)/[subject_tables](https://github.com/datacommonsorg/data/tree/0f069ed0327f95e3f821731aae0d6a00a2565f78/scripts/us_census/acs5yr/subject_tables)/[common](https://github.com/datacommonsorg/data/tree/0f069ed0327f95e3f821731aae0d6a00a2565f78/scripts/us_census/acs5yr/subject_tables/common)/[datacommons_api_wrappers](https://github.com/datacommonsorg/data/tree/0f069ed0327f95e3f821731aae0d6a00a2565f78/scripts/us_census/acs5yr/subject_tables/common/datacommons_api_wrappers)/datacommons_wrappers.py

Can you import that instead?

File: scripts/us_census/acs5yr/subject_tables/common/dc_utils.py (hunk 1 above)
Comment (Abilityguy) [reply scripts/us_census/acs5yr/subject_tables/common/dc_utils.py:35]: Done. Thanks for pointing this out.

File: scripts/us_census/acs5yr/subject_tables/common/dc_utils.py (hunk 2 above)
Comment (Abilityguy) [reply scripts/us_census/acs5yr/subject_tables/common/dc_utils.py]: Done.

File: scripts/us_census/acs5yr/subject_tables/common/dc_utils.py (hunk 3 above)
Comment (Abilityguy) [reply scripts/us_census/acs5yr/subject_tables/common/dc_utils.py]: Thanks. 
I have removed dc_utils.py.

File: scripts/us_census/acs5yr/subject_tables/common/prefetched_outputs/Person_dc_props.json (hunk 4 above)
Comment (Abilityguy) [reply scripts/us_census/acs5yr/subject_tables/common/prefetched_outputs/Person_dc_props.json]: prefetched_outputs is used by `acs_spec_generator_test.py`. Have included a note in `scripts/us_census/acs5yr/subject_tables/common/README.md` on how to refresh them.

**Comment by Abilityguy** [comment]
> Can you import that instead?

Thanks for the pointer. Have imported `datacommons_wrappers.py` and deleted `dc_utils.py`.

**Pull request merged**